vscode/src/completion.ts

```diff
@@ -45,6 +45,15 @@ class QSharpCompletionItemProvider implements vscode.CompletionItemProvider {
       const item = new CompletionItem(c.label, kind);
       item.sortText = c.sortText;
       item.detail = c.detail;
+      item.additionalTextEdits = c.additionalTextEdits?.map((edit) => {
+        return new vscode.TextEdit(
+          new vscode.Range(
+            document.positionAt(edit.range.start),
```

Comment:
Sadly there's another mapping that needs to happen in the JS layer every time we return an offset from the language service. It happens in `npm/src/language-service/language-service.ts`. As an example see how we handle the hover results:

https://github.com/microsoft/qsharp/blob/67bf92acb7fa92ddbdbea05f4c2b66fad92ed985/npm/src/language-service/language-service.ts#L109C13-L109C13

It's easy to miss because the utf8-utf16 conversion only matters when we're using multibyte Unicode characters in the source. Some of our samples (Bernstein-Vazirani I think?) have formulas in the sample that manifest this problem. If you're returning text edits to a position that occurs after these Unicode characters, your edits will be shifted unless you've added the mapping.

Reply:
Nice catch! Unicode can be tricky! Interestingly the implementation for the getHover seemed to be incorrect, as when I copied from that, I got incorrect behavior. After reading the doc for the `mapUtf16UnitsToUtf8Units` I found that the return object is actually a map from the original offset to the new offset value, not an array that takes indices. I've fixed that one as well.